src/main/java/io/netty/incubator/codec/http3/QpackEncoderHandler.java

```diff
@@ -198,4 +217,39 @@ public void channelInactive(ChannelHandlerContext ctx) {
         Http3CodecUtils.criticalStreamClosed(ctx);
         ctx.fireChannelInactive();
     }
+
+    private void handleDecodeFailure(ChannelHandlerContext ctx, QpackException cause, String message) {
+        discard = true;
+        connectionError(ctx, new Http3Exception(QPACK_ENCODER_STREAM_ERROR, message, cause), true);
+    }
+
+    private CharSequence decodeLiteralValue(ByteBuf in, int readerIndex) throws QpackException {
```

Comment:
wouldn't it be better to just call `readerIndex()` before this method and use it for reset if needed ?

Reply:
The caller reads part of the frame before calling this method, so we need to reset it back to the start of the frame which is the passed `readerIndex` here.

Reply:
I think in an "ideal" world the caller would know what to do if we return `null` and rollback.

Reply:
Good call, we are actually already doing that 😄